src/HealthGPS/energy_balance_model.cpp

```diff
@@ -35,66 +46,163 @@ void EnergyBalanceModel::generate_risk_factors([[maybe_unused]] RuntimeContext &
 }
 
 void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
-    auto age_key = core::Identifier{"age"};
-    for (auto &entity : context.population()) {
-        // Ignore if inactive, newborn risk factors must be generated, not updated!
-        if (!entity.is_active() || entity.age == 0) {
+    hgps::Population &population = context.population();
+    const size_t population_size = population.current_active_size();
+    double mean_sim_body_weight = 0.0;
+    double mean_adjustment_coefficient = 0.0;
+
+    // TODO: Compute target body weight.
+    const float target_BW = 100.0;
+
+    // Trial run.
+    for (auto &person : population) {
+        // Ignore if inactive.
+        if (!person.is_active()) {
             continue;
         }
 
-        auto current_risk_factors =
-            get_current_risk_factors(context.mapping(), entity, context.time_now());
-
-        // Model calibrated on previous year's age
-        auto model_age = static_cast<int>(entity.age - 1);
-        if (current_risk_factors.at(age_key) > model_age) {
-            current_risk_factors.at(age_key) = model_age;
-        }
+        auto trial = simulate_person(person, false, 0.0);
+        mean_sim_body_weight += trial[0];
+        mean_adjustment_coefficient += trial[1];
```

Comment:
I think you can do this:

```suggestion
        auto [body_weight, adjustment] = simulate_person(person, false, 0.0);
        mean_sim_body_weight += body_weight;
        mean_adjustment_coefficient += adjustment;
```

Modern C++ ftw.